codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -172,30 +175,31 @@ private void fireChildReadAndRegister(Http2StreamChannel childChannel, Http2Stre
 
     @Override
     public void userEventTriggered(ChannelHandlerContext ctx, Object evt) throws Exception {
-        if (!(evt instanceof Http2StreamStateEvent)) {
+        if (evt instanceof Http2StreamActiveEvent) {
+            Http2StreamActiveEvent activeEvent = (Http2StreamActiveEvent) evt;
+            onStreamActive(activeEvent.streamId(), activeEvent.headers());
+        } else if (evt instanceof Http2StreamClosedEvent) {
+            onStreamClosed(((Http2StreamClosedEvent) evt).streamId());
+        } else {
             ctx.fireUserEventTriggered(evt);
-            return;
         }
+    }
 
-        try {
-            int streamId = ((Http2StreamStateEvent) evt).streamId();
-            if (evt instanceof Http2StreamActiveEvent) {
-                onStreamActive(streamId);
-            } else if (evt instanceof Http2StreamClosedEvent) {
-                onStreamClosed(streamId);
-            } else {
-                throw new UnsupportedMessageTypeException(evt);
+    private void onStreamActive(int streamId, Http2HeadersFrame headersFrame) {
+        final Http2StreamChannel childChannel;
+        if (isOutbound(streamId)) {
+            if (!(headersFrame instanceof ChannelCarryingHeadersFrame)) {
+                throw new IllegalArgumentException("needs to be wrapped");
             }
-        } finally {
-            ReferenceCountUtil.release(evt);
+            childChannel = ((ChannelCarryingHeadersFrame) headersFrame).channel();
+            childChannel.streamId(streamId);
+        } else {
+            ChannelFuture future = bootstrap.connect(streamId);
+            childChannel = (Http2StreamChannel) future.channel();
         }
-    }
 
-    private void onStreamActive(int streamId) {
-        ChannelFuture future = createStreamChannel(ctx, streamId, streamHandler);
-        Http2StreamChannel childChannel = (Http2StreamChannel) future.channel();
-        Http2StreamChannel oldChannel = childChannels.put(streamId, childChannel);
-        assert oldChannel == null;
+        Http2StreamChannel existing = childChannels.put(streamId, childChannel);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "existing". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)